cpp/oneapi/dal/algo/svm/backend/utils.hpp

```diff
@@ -43,105 +45,142 @@ struct binary_label_t {
 };
 
 template <typename Float>
-inline array<Float> convert_labels(const array<Float>& arr_label,
-                                   const binary_label_t<Float>& in_binary_labels,
-                                   binary_label_t<Float>& out_binary_labels) {
+inline daal::data_management::NumericTablePtr convert_labels(
```

Comment:
This function violates SRP now. I'd like to to make it simpler, for example (just to illustrate):

```cpp
inline dal::table convert_binary_labels(const dal::table& labels, 
                                        const binary_label_t<Float>& requested_unique_labels, 
                                        binary_label_t<Float>& previous_unique_labels) {
    previous_unique_labels = get_unique_labels(labels);
    if (previous_unique_labels == binary_label_t<Float>{0, 1} ||
        previous_unique_labels == binary_label_t<Float>{-1, 1}) {
        return labels;
    } else {
        return assign_new_labels(labels, previous_unique_labels, requested_unique_labels); 
    }
}
```
And then implement the functions from next level of abstraction. In SVM, you can use it this way:
``` cpp
  auto new_labels = convert_binary_labels(labels, {-1, 1}, input_unique_labels);
  auto daal_new_labels = convert_to_daal_table<Float>(new_labels);
```